nwbinspector/checks/icephys.py

```diff
@@ -6,5 +6,5 @@
 @register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=IntracellularElectrode)
 def check_intracellular_electrode_cell_id_exists(intracellular_electrode: IntracellularElectrode):
     """Check if the IntracellularElectrode contains a cell_id."""
-    if intracellular_electrode.cell_id is None:
+    if getattr(intracellular_electrode, "cell_id", "") is None:  # Will only be None with PyNWB>=2.1.0
```

Comment:
I understand that this would work, but maybe a better option is to be a bit more explicit, using `get_package_version(name="pynwb") < Version("2.1.0")` within the check function itself.

Reply:
I thought about that but that is precisely the precedence that leads us down a rabbit hole in the future to having to figure out which parts of which tests depend on which version of PyNWB are being used at runtime.

So it's a lot easier on us to set the precedence of just safely checking if an attribute is accessible and do an early return if not, independent of the reason that field might not be accessible.

Also, apologies - I had thought the last commit included an adjustment I made swapping it from `getattr` to `hasattr` but it didn't get pushed, my bad.

Reply:
I don't really understand why you think this approach is better. It feels like this other is more explicit and accomplishes the same thing but I suppose it doesn't matter than much so we can do it this way if you really prefer it